CMakeLists.txt

```diff
@@ -148,10 +148,10 @@ add_library(rmm SHARED
 SET_TARGET_PROPERTIES(rmm PROPERTIES BUILD_RPATH "\$ORIGIN")
 
 target_include_directories(rmm PUBLIC
-  "${THRUST_INCLUDE_DIR}"
+  $<BUILD_INTERFACE:${THRUST_INCLUDE_DIR}>
   "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
```

Comment:
You're right that this is an item that needs more work. The best solution, IMO, would be to use `find_package(CUDAToolKit)` both in the RMM build itself, as well as in the RMM config that it installs. This provides targets for the toolkit, nvtx, etc. However, this is unfortunately cmake 3.17+ only, and I didn't want to mess with the existing build too much in the first place in my initial go at this.

Having the absolute path hard-coded there shouldn't usually cause troubles -- when RMM is built with conda, it'll hardcode the path to the corresponding cuda installation, and that won't hurt. The case where things might go wrong is if someone build and installs RMM using one version of CUDA, and then includes that RMM into a downstream project but switches to a different CUDA version to build that project.

Reply:
I'm more fearful of RMM being built in a conda build environment and found in a different conda environment as those would have different CUDA paths.

Similarly different Linux distros have different paths for where the CUDA toolkit gets installed.

We've been meaning to upgrade to the latest CUDA features in CMake anyway where that may be a precursor to being able to handle this gracefully.

Reply:
I think the question is, do you want force users to upgrade to cmake 3.17? That'd of course be the easiest, but while cmake is at 3.18, I think 3.17 isn't exactly in widespread use. The alternative is to backport `FindCUDAToolKit.cmake` (which probably means just making a copy) and include it within the rmm distribution, which shouldn't be too difficult.

Reply:
I think we're okay forcing users to use a newer cmake since it's easily installed in user space. In general RAPIDS is built on top of the open source ecosystem and moves pretty fast along with the ecosystem.

Reply:
I opened #474 to do some prep stuff first, including updating the required cmake version, so I guess that'll show how much disruption requiring 3.17 will cause. I think it can be relaxed relatively easily by including `FindCUDAToolkit.cmake` in the RMM tree, if needed.